crates/ruff/src/rules/pyupgrade/snapshots/ruff__rules__pyupgrade__tests__UP006.py.snap

```diff
@@ -19,13 +19,13 @@ UP006.py:4:10: UP006 [*] Use `list` instead of `List` for type annotations
 6 6 | 
 7 7 | 
 
-UP006.py:11:10: UP006 [*] Use `list` instead of `List` for type annotations
+UP006.py:11:10: UP006 [*] Use `list` instead of `typing.List` for type annotation
```

Comment:
The recommendation here doesn't match the code (the code only uses `List` not `typing.List`) which can be confusing to users because they don't recognize their own code.

Reply:
rustc switches depending on whether symbol is ambiguous:

Different names:

```rust
fn a(b: String) -> String {
    return b;
}

fn main() {
    struct String2;
    let x = a(String2);
    println!("Hello, {}!", x);
}
```

```
error[E0308]: mismatched types
 --> src/main.rs:7:15
  |
7 |     let x = a(String2);
  |             - ^^^^^^^ expected `String`, found `String2`
  |             |
  |             arguments to this function are incorrect
  |
note: function defined here
 --> src/main.rs:1:4
  |
1 | fn a(b: String) -> String {
  |    ^ ---------
```

```rust
fn a(b: String) -> String {
    return b;
}

fn main() {
    struct String;
    let x = a(String);
    println!("Hello, {}!", x);
}
```

Same names:

```
error[E0308]: mismatched types
   --> src/main.rs:7:15
    |
7   |     let x = a(String);
    |             - ^^^^^^ expected `std::string::String`, found `main::String`
    |             |
    |             arguments to this function are incorrect
    |
    = note: `main::String` and `std::string::String` have similar names, but are actually distinct types
note: `main::String` is defined in the current crate
   --> src/main.rs:6:5
    |
6   |     struct String;
    |     ^^^^^^^^^^^^^
note: `std::string::String` is defined in crate `alloc`
   --> /home/konsti/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/string.rs:365:1
    |
365 | pub struct String {
    | ^^^^^^^^^^^^^^^^^
note: function defined here
   --> src/main.rs:1:4
    |
1   | fn a(b: String) -> String {
    |    ^ ---------
```